classes/submission/Collector.inc.php

```diff
@@ -194,6 +194,16 @@ public function offset(int $offset): AppCollector
         return $this;
     }
 
+    /**
+     * Remove any limit and offset restrictions
+     */
+    public function unlimited(): AppCollector
```

Comment:
I think we've talked about this before, but IMO these calls should be unlimited by default. I think the API is the only use case where a limit is normal, so the limit should be introduced there. Otherwise I think it's likely that limits will be incorrectly imposed where they aren't expected, and our test data sets will be shallow enough that the problem won't show up until production.

Reply:
They are unlimited by default. In this case, the method is used when you want to take a previously limited collector and remove the limit options on it. Since the limit method requires an integer, `limit(int $limit)`, we can't pass in a `null` value to remove the limitation.

This is a little ackward, but the use-case is in the API handlers (see [here](https://github.com/pkp/pkp-lib/pull/7198/files#diff-5793b9beeca15a272911de3b42176b7d4e2dbd2ad3ba9e2c5fb8cb79b5fb8b44)). Basically, we reuse the collector to get the total possible items (for pagination). So it goes something like this:

```php
$collector = ...;
foreach ($slimRequest->getQueryParams() as $param => $val) {
  switch ($param) {
    case 'isPublished':
      $collector->isPublished(true);
      break;
    case 'limit':
      $collector->limit((int) $val);
      break;
    case 'offset':
      $collector->offset((int) $val);
      break;
    ...
  }
}

$submissions = Repo::submission()->getMany($collector);
$totalItems =  Repo::submission()->getMany($collector->unlimited());
```

Maybe it's better to only apply these query parameters later, but it also felt akward to split things up.

```php
$collector = ...;
foreach ($slimRequest->getQueryParams() as $param => $val) {
  switch ($param) {
    case 'isPublished':
      $collector->isPublished(true);
      break;
    ...
  }
}

$totalItems =  Repo::submission()->getMany($collector->unlimited());

if (isset($param['count'])) {
  $collector->limit((int) $param['count']);
}
if (isset($param['offset'])) {
  $collector->limit((int) $param['offset']);
}

$submissions = Repo::submission()->getMany($collector);
```

I'm comfortable with either approach, but it does mean processing the `count` and `offset` query params in separate blocks of code, so care would need to be taken to ensure they receive the same sanitization as the other params.

Reply:
Ah, gotcha. I thought we were still hard-coding a `30` limit in the collector, and would be happy to see that gone.

Yes, I've hit a similar use case for both counting and querying and for the moment executed the counting query before the getMany. But I like keeping things together as you've done. As a general practice, how do you feel about permitting nulls in filter functions in order to reset the constraint? So it would be...
```
class Collector
{
    public $someIds = null; // Null means we don't care

    public function filterBySomeIds(array? $someIds): self
    {
        $this->someIds = $someIds;
        return $this;
    }
}
```
The only change from what we're doing is the `?` in the `array?` typehint in `filterBy...`, and no need for a new set of functions to reset each constraint.

Reply:
I think that makes sense. :+1: 

I'll try to work this into my stuff next week and then document it in #pkp-dev for the group.